src/display/annotation_layer.js

```diff
@@ -315,6 +335,16 @@ var TextAnnotationElement = (function TextAnnotationElementClosure() {
 
   Util.inherit(TextAnnotationElement, AnnotationElement, {
     /**
+     * Check does this annotation need HTML element.
+     *
+     * @public
+     * @memberof TextAnnotationElement
+     */
+    hasHtml: function TextAnnotationElement_hasHtml() {
+      return this.data.hasPopup || this.data.title || this.data.contents;
```

Comment:
Is this really equivalent to the current code?
Because currently in the `hasPopup` case, whether or not `hasHtml === true` is determined by the `title`/`contents` of the Popup *itself*, and not of the TextAnnotation. Won't this line incorrectly return true in some cases?